### PR TITLE
Introduce ordered quantization ops for the CUDA EP [1/n]

### PR DESCRIPTION
**Pull request by hariharans29**
**Description**:  This PR introduces the first set of schemas, kernels, and tests for ordered quantization ops for use in the CUDA EP. The kernels were mostly developed by @zhanghuanrong with some minor enhancements / tests contributed across the team (true for upcoming similar PRs)

The fusion logic to fuse QDQ subgraphs into ordered ops will be contributed to the transformer optimization tools suite once the kernels have been merged (https://github.com/microsoft/onnxruntime/pull/12661)

### CODE REVIEW

File: onnxruntime/contrib_ops/cuda/quantization/qordered_ops/qordered_layer_norm_impl.cu [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+// TODO: Const Ref ?
+__device__ inline char4 QuantizeFloat4Char4(const float4 val4, const float rscale) {
+  return char4{QuantizeFloatS8(val4.x, rscale), QuantizeFloatS8(val4.y, rscale),
+               QuantizeFloatS8(val4.z, rscale), QuantizeFloatS8(val4.w, rscale)};
+}
+
+// TODO: Const Ref ?
+__device__ inline int32_t Dp4a_Defined(const char4 input_1, const char4 input_2) {
+  return static_cast<int8_t>(input_1.x) * static_cast<int8_t>(input_2.x) +
```
Comment (hariharans29) [inline onnxruntime/contrib_ops/cuda/quantization/qordered_ops/qordered_layer_norm_impl.cu:27]: Need to manually define a dot product function as the dp intrinsics are only available for arch >= SM61

File: onnxruntime/test/contrib_ops/qordered_matmul_op_test.cc [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+static void RunQOrdered_MatMul_Test(
+    std::vector<int64_t> const& shape_A,
+    std::vector<int64_t> const& shape_B,
+    std::vector<int64_t> const& shape_Y,
+    OrderCublasLt weight_order,
+    float scale_A, float scale_B, float scale_C, float scale_Y,
+    bool add_bias = false, bool broadcast_c_batch = false) {
+  // Needs Turing or higher architecture
+  if (NeedSkipIfCudaArchLowerThan(750)) {
```
Comment (hariharans29) [inline onnxruntime/test/contrib_ops/qordered_matmul_op_test.cc:20]: Can't run these tests on CI as the IMMA MatMul needs Turing or higher arch to run

File: tools/ci_build/amd_hipify.py [hunk 3]
```diff
@@ -47,6 +47,19 @@
     "quantization/attention_quantization_impl.cu",
     "quantization/attention_quantization_impl.cuh",
     "quantization/quantize_dequantize_linear.cc",
+    "quantization/qordered_ops/qordered_common.cuh",
```
Comment (hariharans29) [inline tools/ci_build/amd_hipify.py:50]: Can't hipify these files just yet

**Review by zhanghuanrong — APPROVED** [review]
LGTM

**Pull request merged**